main/src/ca/uwaterloo/flix/tools/Packager.scala

```diff
@@ -1,23 +1,51 @@
 package ca.uwaterloo.flix.tools
 
-import java.io.PrintWriter
-import java.nio.file.attribute.BasicFileAttributes
-import java.nio.file._
-import java.util.zip.{ZipEntry, ZipFile, ZipOutputStream}
-
 import ca.uwaterloo.flix.api.Flix
 import ca.uwaterloo.flix.language.ast.Ast.Source
 import ca.uwaterloo.flix.runtime.CompilationResult
-import ca.uwaterloo.flix.util.vt.TerminalContext
+import ca.uwaterloo.flix.tools.github.GitHub
 import ca.uwaterloo.flix.util._
+import ca.uwaterloo.flix.util.vt.TerminalContext
 
+import java.io.PrintWriter
+import java.nio.file._
+import java.nio.file.attribute.BasicFileAttributes
+import java.util.zip.{ZipEntry, ZipFile, ZipOutputStream}
 import scala.collection.mutable
 
 /**
   * An interface to manage flix packages.
   */
 object Packager {
 
+  /**
+    * Installs a flix package from the Github `project`.
+    *
+    * `project` must be of the form `<owner>/<repo>`
+    *
+    * The package is installed at `lib/<owner>/<repo>`
+    */
+  def install(project: String, p: Path, o: Options)(implicit tc: TerminalContext): Unit = {
+    val proj = GitHub.parseProject(project)
+    val release = GitHub.getLatestRelease(proj)
+    val assets = release.assets.filter(_.name.endsWith(".fpkg"))
+    val lib = getLibraryDirectory(p)
+    val assetFolder = lib.resolve(proj.owner).resolve(proj.repo)
+
+    // create the asset directory if it doesn't exist
+    Files.createDirectories(assetFolder)
+
+    // clear the asset folder
+    assetFolder.toFile.listFiles.foreach(_.delete())
```

Comment:
Hmm. This makes me nervous. 

I don't like deleting things. I am okay with overwriting but even then I usually perform a check.

For example, the backend will only overwrite a `.class` file if it starts with the class bytecode signature (CAFEBABE).

Can we do something similar?

Reply:
A basic check would be something like:
* has the `.fpkg` extension, and
* is a zipfile (I assume it has a special header)

Should we just throw an exception if we run into a file not matching these criteria?

Reply:
> 
> 
> A basic check would be something like:
> 
>     * has the `.fpkg` extension, and
> 
>     * is a zipfile (I assume it has a special header)
> 
> 
> Should we just throw an exception if we run into a file not matching these criteria?

Yes. I would much prefer that.